src/common/acl_principal.c

```diff
@@ -309,3 +311,116 @@ daos_acl_principal_to_gid(const char *principal, gid_t *gid)
 	D_FREE(buf);
 	return rc;
 }
+
+static bool
+is_special_type(enum daos_acl_principal_type type)
+{
+	switch (type) {
+	case DAOS_ACL_OWNER:
+	case DAOS_ACL_OWNER_GROUP:
+	case DAOS_ACL_EVERYONE:
+		return true;
+	case DAOS_ACL_USER:
+	case DAOS_ACL_GROUP:
+	default:
+		break;
+	}
+
+	return false;
+}
+
+static int
+get_principal_type_from_str(const char *principal_str,
+			    enum daos_acl_principal_type *type)
+{
+	if (strncmp(principal_str, DAOS_ACL_PRINCIPAL_OWNER,
```

Comment:
I think user and group are most likely, and owner and owner-group are least likely. Will move them around. This will be a relatively rare event triggered by a user command to edit an ACL, so I suspect it doesn't save many cycles to key off the first character.